src/sgame/sg_cmds.cpp

```diff
@@ -2045,6 +2045,11 @@ bool G_AlienEvolve( gentity_t *ent, class_t newClass, bool report, bool dryRun )
 		return false;
 	}
 
+	if ( newClass == currentClass )
```

Comment:
do you need to respect `report` and report back soemthign to the user?

Reply:
I discussed this on IRC with @illwieckz:
```
<+cu-kai> anyway: that commit silently prevents it
<+cu-kai> (do we really need to feed back for this?)
<+illwieckz> no
<+illwieckz> evolving to itself seems to be a bug to me
<+cu-kai> i agree
<+cu-kai> so simply fail silently, instead of "You cannot evolve to your current class"?
<+cu-kai> if so: i can make a PR for this now if you like
<+illwieckz> yes please
```

Reply:
Then please add a comment about this in the code.